pkg/restapi/operation/operations.go

```diff
@@ -127,20 +130,36 @@ Request body: %s`, string(requestBody))
 	c.createDataVault(rw, &config, req.Host, configBytesForLog)
 }
 
+// Generates a vault name and a vaultID. The vault name is for internal usage to bypass couchdb's database naming
+// convention, and vaultID is a base58-encoded random value that's visible to the user.
 func (c *Operation) createDataVault(rw http.ResponseWriter, config *models.DataVaultConfiguration, hostURL string,
 	configBytesForLog []byte) {
 	if config.ReferenceID == "" {
 		writeBlankReferenceIDErrMsg(rw, configBytesForLog)
 		return
 	}
 
-	err := c.vaultCollection.createDataVault(config.ReferenceID)
+	dbVaultName := uuid.New().String()
```

Comment:
You can skip the mapping thing if you just base58 decode the vault ID (below) back into a UUID.